packages/data-schema/src/SchemaProcessor.ts

```diff
@@ -1316,6 +1324,168 @@ const mergeCustomTypeAuthRules = (
   }
 };
 
+/**
+ * Generates input types for custom operations in the schema.
+ *
+ * Processes operation arguments to create corresponding input types,
+ * handling referenced and inline custom types, enums, and nested structures.
+ * Manages circular references and prevents duplicate processing.
+ *
+ **/
+
+function generateInputTypes(
+  operationName: string,
+  args: Record<string, any>,
+  getRefType: (name: string, referrerName?: string) => GetRef,
+): {
+  inputTypes: { name: string; fields: Record<string, any> }[];
+  argDefinitions: string[];
+  collectedEnums: Map<string, any>;
+} {
+  const inputTypes: {
+    name: string;
+    fields: Record<string, any>;
+  }[] = [];
+  const argDefinitions: string[] = [];
+  const collectedEnums: Map<string, any> = new Map();
+  const processedTypes = new Set<string>(); // Track processed types to avoid duplicates
```

Comment:
This function has grown a bit out of hand. I tried breaking it apart into a few shorter helpers, but the way state is being shared/reused to compose the results doesn't lend itself to that kind of decomposition.

I think this deserves to be an object where the shared / accrued state can be camptured in the instance and the methods are free to call each other. It could be 2 tiers of objects depending on how it fits together.

All of this is better suited to being looked at in the refactor of this code where any of these changes would be a substantial departure from the way the schema processor is structured today.

Recommend we do not refactor this in this change, but capturing some thoughts I'm having as something to refer back to for the refactor.